Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -420,6 +431,12 @@ namespace AzToolsFramework
         }
     }
 
+    void DocumentPropertyEditor::ReleaseHandler(AZStd::unique_ptr<PropertyHandlerWidgetInterface>&& handler)
```

Comment:
shouldn't need any of this now that I fixed my bug...

Reply:
Returning handlers to the pool is 100% necessary because handlers can trigger patches that remove themselves (ExampleAdapter ran into this with the "-" button on its tree structure) which causes Bad Times.

This also is a nudge in the direction of reusing handlers in a pool, if/when that's ever necessary from a perf standpoint.